render/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/render/jpql/introspector/impl/JavaxJpqlIntrospector.kt

```diff
@@ -20,6 +24,21 @@ class JavaxJpqlIntrospector : JpqlIntrospector {
             null
         }
     }
+
+    override fun introspect(property: KCallable<*>): JpqlPropertyDescription? {
+        return when (property) {
+            is KProperty1<*, *> -> JpqlProperty(property.name)
+            is KFunction1<*, *> -> JpqlProperty(resolvePropertyName(property))
+            else -> null
+        }
+    }
+
+    private fun resolvePropertyName(getter: KFunction1<*, *>): String =
+        if (getter.name.startsWith("is")) {
+            getter.name
+        } else {
+            getter.name.removePrefix("get").replaceFirstChar { it.lowercase() }
+        }
```

Comment:
I suppose the Property Introspector can be separated from `Javax` and `Jakarta`. 

`JpqlIntrospector` is designed to delegate to another `Intropsector` if it cannot introspect itself. Therefore, I think we can reduce the duplication by creating a separate class that only introspects properties and adding it to the `Context` together. 

```kotlin
class JavaxJpqlEntityIntrospector : JpqlIntrospector {
    override fun introspect(type: KClass<*>): JpqlEntityDescription? {
        // ...
    }

    override fun introspect(property: KCallable<*>): JpqlPropertyDescription? {
        return null
    }
}

class JakartaJpqlEntityIntrospector : JpqlIntrospector {
    override fun introspect(type: KClass<*>): JpqlEntityDescription? {
        // ...
    }

    override fun introspect(property: KCallable<*>): JpqlPropertyDescription? {
        return null
    }
}

class KotlinStyleJpqlPropertyIntrospector : JpqlIntrospector {
    override fun introspect(type: KClass<*>): JpqlEntityDescription? {
        return null
    }

    override fun introspect(property: KCallable<*>): JpqlPropertyDescription? {
        return when (property) {
            is KProperty1<*, *> -> JpqlProperty(property.name)
            is KFunction1<*, *> -> JpqlProperty(resolvePropertyName(property))
            else -> null
        }
    }
    
    // ...
}

private class DefaultModule : JpqlRenderModule {
    override fun setupModule(context: JpqlRenderModule.SetupContext) {
        // ...

        context.appendIntrospector(KotlinStyleJpqlPropertyIntrospector)

        // ...
    }
}
```

I think also you could create two abstract classes, `JpqlEntityIntrospector` and `JpqlPropertyIntrospector`, that implement JpqlIntropsector.